### ISSUE TITLE
Add linearize method for blackbox systems

### ISSUE THREAD

**Issue by ueliwechsler** [JuliaReach/MathematicalSystems.jl#161]
I could be a useful feature to have a `linearize` method for black-box systems which does compute the linearization at a certain point `(x,u,w)` for an arbitrary function `f(x,u,w)`.

Using the fields `statedim`, `inputsdim` and possibly `noisedim`. 

To avoid any precompilation overhead, we should use `Requires` to load the used AD library.

**Comment by mforets** [comment]
I agree to add a linearization routine, ideally through existing tools as you propose. I have previously written some code that was using TaylorSeries for the computations, but i would have to clean it up.
